walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,108 +14,255 @@
 
 package com.google.sps;
 
+import java.util.Arrays;
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.HashSet;
+import java.util.stream.Collectors;
 
 public final class FindMeetingQuery {
-    
+  
+  /** 
+   * First Solution: 
+   * Separate {@code events} in two lists based on attendee type.
+   * Call .findOpenTimeRanges() on each list
+   * Return a collection with valid time slots accroding to requesst.
+   *
+   */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+
     // If the request has no attendees, the open time slot is the entire day.
     if (request.getAttendees().isEmpty() && 
         request.getOptionalAttendees().isEmpty()) {
         return Arrays.asList(TimeRange.WHOLE_DAY);
     }
 
     // Sort {@code events} by starting time.
-    List<Event> eventsList = new ArrayList(events);
-    eventsList.sort((Event a, Event b) -> {
-        return TimeRange.ORDER_BY_START.compare(a.getWhen(), b.getWhen());
-    });
+    List<Event> eventsList = new ArrayList<>(events);;
+
+    
 
     // Filter out events with attendees not in {@code request}.
-    if (request.getOptionalAttendees() != null){ 
     eventsList = eventsList.stream()
-                           .filter(e -> request.getAttendees()
-                                               .containsAll(e.getAttendees()) || 
-                                        request.getOptionalAttendees()
-                                               .containsAll(e.getAttendees()))
+                           .filter(event -> !Collections.disjoint(request.getAttendees(), event.getAttendees()) || 
+                                        !Collections.disjoint(request.getOptionalAttendees(), event.getAttendees()))
                            .collect(Collectors.toList());
-    } else {
-        eventsList = eventsList.stream()
-                           .filter(e -> request.getAttendees()
-                                               .containsAll(e.getAttendees()))
-                           .collect(Collectors.toList());
-    }
+
+    eventsList.sort((Event a, Event b) -> {
+        return TimeRange.ORDER_BY_START.compare(a.getWhen(), b.getWhen());
+    });
+    
     List<Event> mandatoryEvents = eventsList.stream()
```

Comment:
I tried that and it caused one of the tests with optional attendees to fail because even if allTimeRanges has entries and the request attendees is empty, you would still have to find the time slots with the maximum number of optional attendees able to go.